components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -259,9 +348,20 @@ protected void processAuthenticationResponse(HttpServletRequest request,
             String tokenEndPoint = getTokenEndpoint();
             String fbAuthUserInfoUrl = getUserInfoEndpoint();
 
-            String callbackUrl = getCallbackUrl(authenticatorProperties);
-            String code = getAuthorizationCode(request);
-            String token = getToken(tokenEndPoint, clientId, clientSecret, callbackUrl, code);
+            String token;
+            if (isTrustedTokenIssuer(context) && isNativeSDKBasedFederationCall(request)) {
+                String idToken = request.getParameter(ID_TOKEN_PARAM);
+                token = request.getParameter(ACCESS_TOKEN_PARAM);
+                try {
+                    validateJWTToken(context, idToken);
+                } catch (ParseException | IdentityOAuth2Exception | JOSEException e) {
+                    throw new AuthenticationFailedException("JWT Token validation Failed.");
```

Comment:
Are these exceptions client exceptions or server exceptions? If all these exceptions are client exceptions let's add a debug log printing 'e'. Otherwise let's throw 'e' along with the message.

Reply:
These errors can occur due to sending a wrong ID token from the client side or due to incorrectly configuring the IdP from the server side.

Reply:
Addressed by https://github.com/wso2-extensions/identity-outbound-auth-facebook/pull/54/commits/2be2f96763127c69f9b8f450a9c61c662dd6b5b1